### PR TITLE
fix s3 model training and loading

### PR DESCRIPTION
**Pull request by DominicBreuker**
Adds a fix for issue #633 (S3 persistence for models not working in 0.10.1). The issue is that models persistent in S3 are not loading on server startup.

**Proposed changes**:
- most important: update `rasa_nlu/persistor.py` to use API of boto3 (Seems to currently call methods from boto2 but boto3 is installed as per requirements. Compare http://boto.cloudhackers.com/en/latest/ref/s3.html#boto.s3.bucket.Bucket.list vs http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Bucket.objects).
- extracts constant DEFAULT_PROJECT_NAME to `RasaNLUConfig` to make it accessible in both `rasa_nlu/data_router.py` and `rasa_nlu/persistor.py`. Required for loading models for project "default" since currently, it tried to load S3 objects whose key starts with "None_" but training creates objects with keys prefixed "default_".
- small fixes for lazy model loading in `rasa_nlu/project.py`: avoid error in absolute path construction if no path is given (happens for cloud storage) and ensures locks are released if loading fails (currently, server blocks after fail and must be restarted)

Plz let me know if you would like to merge this and if yes, how and what needs to change for that to happen. Would love to avoid having to maintain a fork :)

p.s.: I ran tests inside a container based off of `docker/Dockerfile_devtest` (included in PR for reference, started with `docker run -it --rm -v `pwd`/rasa_nlu:/app/rasa_nlu  <image_name> pytest tests/`). Tests said `148 passed` before and after my changes, so I hope I did not break anything important.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog

### CODE REVIEW

File: rasa_nlu/persistor.py [hunk 1]
```diff
@@ -143,10 +143,11 @@ def __init__(self, aws_region, bucket_name, endpoint_url):
     def list_models(self, project):
         # type: (Text) -> List[Text]
         try:
-            blob_iterator = self.bucket.list(
-                    prefix=self._project_prefix(project))
-            return [self._project_and_model_from_filename(b.name)[1]
-                    for b in blob_iterator]
+            prefix = self._project_prefix(project)
+            model_objects = [str(b.key) for b in self.bucket.objects.all()
```
Comment (tmbo) [inline rasa_nlu/persistor.py]: need to import `from builtins import str` (otherwise this fails on py3)

Comment (tmbo) [reply rasa_nlu/persistor.py]: how about using `self.bucket.objects.filter(Prefix=prefix)`?

File: docker/Dockerfile_devtest [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM rasa/rasa_nlu:latest-full
```
Comment (tmbo) [inline docker/Dockerfile_devtest]: I don't think we need this container, do we?

**Comment by tmbo** [comment]
And please update the changelog (`CHANGELOG.rst`) to include the fix

**Comment by DominicBreuker** [comment]
all right :)
- Changelog updated
- Dockerfile removed 
- Used the filter method. Much better!

**Comment by tmbo** [comment]
Looking very good :tada:

**Pull request merged**